Library/Homebrew/command.rb

```diff
@@ -0,0 +1,133 @@
+module Homebrew
+  class Command
+    attr_accessor :command_name, :description
+    attr_reader :valid_options, :help_output, :man_output
+
+    def initialize_variables
+      @valid_options = []
+      @argv_tokens = ARGV.dup.uniq
+    end
+
+    def options(&block)
+      initialize_variables
+      instance_eval(&block)
+      generate_documentation
+    end
+
+    def option(option_name, **option_hash, &block)
+      option_hash[:option_name] = "--#{option_name}"
```

Comment:
I'm inclined to say that we should just include the `--` and `-` in the options as passed in the DSL, instead of constructing them here:

```ruby
option "--foo", desc: "foo description"
```

That would make us more consistent with the style used by OptionParser and others.